src/common.cpp

```diff
@@ -1873,6 +1884,63 @@ std::string get_executable_path(const char *argv0) {
     } else {
         return std::string(buff);
     }
+#elif defined(__OpenBSD__)
+    static std::string result;
+    if (!result.empty()) return result;
+    int cntp = 0;
+    struct stat st;
+    bool ok = false;
+    std::string path;
+    kinfo_file *kif = nullptr;
+    static kvm_t *kd = nullptr;
+    char errbuf[_POSIX2_LINE_MAX];
+    const char *pwd = nullptr, *penv = nullptr;
+    if (!std::string(argv0).empty()) {
```

Comment:
Like I said, argv0 should just be saved as a string, so you don't have to reconstruct it multiple times.

E.g.

```c++
std::string strargv0 = argv0;

if (strargv0.empty()) {

// ...

        if (strargv0[0] == '/') {
            path = argv0;
        } else if (strargv0.find('/') == std::string::npos) {